java-test-samples/async-lambda-dynamodb/template.yaml

```diff
@@ -0,0 +1,144 @@
+        ServerSideEncryptionConfiguration:
+        - ServerSideEncryptionByDefault:
+            SSEAlgorithm: AES256
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: true
+        BlockPublicPolicy: true
+        IgnorePublicAcls: true
+        RestrictPublicBuckets: true
+
+  DestinationBucket:
+    Type: AWS::S3::Bucket
+    UpdateReplacePolicy: Delete
+    Properties:
+      BucketName:
+        !Sub "async-lambda-dynamodb-destination-${AWS::AccountId}"
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+        - ServerSideEncryptionByDefault:
+            SSEAlgorithm: AES256
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: true
+        BlockPublicPolicy: true
+        IgnorePublicAcls: true
+        RestrictPublicBuckets: true
+
+  RecordTrasnformationTable:
+    Type: AWS::DynamoDB::Table
+    Condition: CreateTestResources
+    Properties:
+      TableName:
+        !Sub "async-lambda-dynamodb-record-transformation-${AWS::AccountId}"
+      AttributeDefinitions:
+        - AttributeName: id
+          AttributeType: S
+      KeySchema:
+        - AttributeName: id
+          KeyType: HASH
+      ProvisionedThroughput:
+        ReadCapacityUnits: 2
+        WriteCapacityUnits: 2
+
+  ToUpperCaseHandlerFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Runtime: java11
+      Handler: com.amazon.aws.sample.ToUpperCaseHandler::handleRequest
+      Timeout: 60
+      MemorySize: 1024
+      Environment:
+        Variables:
+          JAVA_TOOL_OPTIONS: -XX:+TieredCompilation -XX:TieredStopAtLevel=1
+          DESTINATION_BUCKET_NAME: !Sub "async-lambda-dynamodb-destination-${AWS::AccountId}"
+      Policies:
+        - S3ReadPolicy:
+            BucketName: 
+              !Sub "async-lambda-dynamodb-source-${AWS::AccountId}"
+        - S3CrudPolicy:
+            BucketName: 
+              !Sub "async-lambda-dynamodb-destination-${AWS::AccountId}"
+      Events:
+        FileUpload:
+          Type: S3
+          Properties:
+            Bucket: !Ref SourceBucket
+            Events: s3:ObjectCreated:*
+            Filter:
+              S3Key:
+                Rules:
+                  - Name: suffix
+                    Value: '.txt'
+
+
+  TransformationHandlerFunction:
+    Type: AWS::Serverless::Function
+    Condition: CreateTestResources
+    Properties:
+      Runtime: java11
+      Handler: com.amazon.aws.sample.TransformationHandler::handleRequest
+      Timeout: 60
+      MemorySize: 1024
+      Environment:
+        Variables:
+          JAVA_TOOL_OPTIONS: -XX:+TieredCompilation -XX:TieredStopAtLevel=1
+          RECORD_TRANSFORMATION_TABLE_NAME: !Sub "async-lambda-dynamodb-record-transformation-${AWS::AccountId}"
+      Policies:
+        - S3ReadPolicy:
+            BucketName: 
+              !Sub "async-lambda-dynamodb-destination-${AWS::AccountId}"
+        - DynamoDBWritePolicy:
+            TableName: !Ref RecordTrasnformationTable
+      Events:
```

Comment:
What are we trying to test here? S3 => Lambda => S3 => Lambda?

Reply:
Added information to the README.md